weaviate/connect/connection.py

```diff
@@ -110,15 +113,25 @@ def _create_session(self, auth_client_secret: Optional[AuthCredentials]) -> None
         ValueError
             If no authentication credentials provided but the Weaviate server has OpenID configured.
         """
+        oidc_url = self.url + self._api_version_path + "/.well-known/openid-configuration"
         response = requests.get(
-            self.url + self._api_version_path + "/.well-known/openid-configuration",
+            oidc_url,
             headers={"content-type": "application/json"},
             timeout=self._timeout_config,
             proxies=self._proxies,
         )
         if response.status_code == 200:
-            if auth_client_secret is not None:
+            # Some setups are behind proxies that return some default page - for example a login - for all requests.
+            # If the response is not json, we assume that this is the case and try unauthenticated access. Any auth
+            # header provided by the user is unaffected.
+            try:
                 resp = response.json()
+            except JSONDecodeError:
+                _Warnings.auth_cannot_parse_oidc_config(oidc_url)
+                self._session = requests.Session()
+                return
```

Comment:
Why do we assume that is the response is not decodable it works with authorization header passed by user?

Reply:
We assume that there is no weaviate authentication if we cannot parse it, doesn't matter if users provide any header or not. Is there anything better to do?

Reply:
If there is no Weaviate authentication Weaviate returns 404 status code for `/.well-known/openid-configuration`. At least it was like this.

Reply:
Check the linked issue. :)

The user has a proxy that returns a response for all URLs that do not exists, eg every request returns status code 200

Reply:
Sorry, missed the Issue there. Makes sense. Then it can be merged.

Reply:
thanks!